### PR TITLE
Dynamic stubbing

### PR DESCRIPTION
**Pull request by jakub-dzon**
Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

File: internal/app/handlers/dynamic/stubbing-provider.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	"net/http"
+	"strings"
+
+	"github.com/machacekondra/fakeovirt/internal/app/router"
+
+	"github.com/gorilla/pat"
+	"github.com/machacekondra/fakeovirt/api/stubbing"
+)
+
+//
```
Comment (pkliczewski) [inline internal/app/handlers/dynamic/stubbing-provider.go]: Empty comment?

File: internal/app/handlers/dynamic/stubbing-provider.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	delegate.Post("/stub", h.Stub())
+	delegate.Post("/reset", h.Reset())
+}
+
+// Reset restores original paths in the router held by the StubbingProvider
+func (h *StubbingProvider) Reset() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		newRouter := h.purge()
+
+		h.configure()
```
Comment (pkliczewski) [inline internal/app/handlers/dynamic/stubbing-provider.go:57]: This looks similar to logic in `Configure`. Is it possible to reuse it?

**Review by pkliczewski — COMMENTED** [review]
Please make sure that it would still work with CDI.

File: internal/app/handlers/dynamic/stubbing-provider.go (hunk 1 above)
Comment (jakub-dzon) [reply internal/app/handlers/dynamic/stubbing-provider.go]: Thanks!

File: internal/app/handlers/dynamic/stubbing-provider.go (hunk 2 above)
Comment (jakub-dzon) [reply internal/app/handlers/dynamic/stubbing-provider.go:57]: Taking into account the differences between `Configure` and `Reset` (especially conditional branching in the latter) I don't think that there is a way to extract more common elements.

**Comment by jakub-dzon** [comment]
> Please make sure that it would still work with CDI.

:+1: good idea. Will do.

File: internal/app/router/delegating-router.go.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package router
```
Comment (machacekondra) [inline internal/app/router/delegating-router.go.go]: The name of the file has two .go suffixes.

Comment (jakub-dzon) [reply internal/app/router/delegating-router.go.go]: Thanks! :facepalm:

**Review by machacekondra — APPROVED** [review]
Looks cool, thanks!

**Pull request merged**